### ISSUE TITLE
Share sheet icons on Android Oreo

### ISSUE THREAD

**Issue by ming13** [BranchMetrics/android-branch-deep-linking#478]
Android version: 8.0.0
Branch SDK version: 2.11.1

We are using the sharing sheet and icons there are quite large. [Take a look at the screenshot](https://user-images.githubusercontent.com/200401/30472966-3cf3b71a-9a07-11e7-9356-062dea475ca3.png).

**Comment by EvangelosG** [comment]
Hey @ming13 -- Thanks a lot for reporting this. It looks like I'll need to upgrade our SDK to recognize and handle adaptive icons accordingly. I will dig into this and ideally push an update next week. In the meantime you can play with the share sheet icon rendering code [here](https://github.com/BranchMetrics/android-branch-deep-linking/blob/65f8c34ccc6705331b50348f99a66a13da19cf8c/Branch-SDK/src/io/branch/referral/ShareLinkManager.java#L419).

**Comment by EvangelosG** [comment]
Hey @ming13 -- It'll be a bit improved before implementation, and I need to add a builder method. But if you would like to correct your icon sizes now, here's the code. First you'll need this function:

```
public static int dpToPx(Context context, int dp) {
        DisplayMetrics displayMetrics = context.getResources().getDisplayMetrics();
        return Math.round(dp * (displayMetrics.xdpi / DisplayMetrics.DENSITY_DEFAULT));
    }
```

Then, right below [this else statement](https://github.com/BranchMetrics/android-branch-deep-linking/blob/65f8c34ccc6705331b50348f99a66a13da19cf8c/Branch-SDK/src/io/branch/referral/ShareLinkManager.java#L416) you can resize the Drawable:

```
int size = BranchUtil.dpToPx(context_, 50);
appIcon.setBounds(0, 0, size, size);
this.setTextAppearance(context_, android.R.style.TextAppearance_Medium);
this.setCompoundDrawables(appIcon, null, null, null);
viewItemMinHeight_ = Math.max(viewItemMinHeight_, (appIcon.getIntrinsicHeight() + padding));
```

**Comment by ming13** [comment]
Thanks! But I think we’ll wait for the next release ;-)

**Comment by sojanpr** [comment]
@ming13 this fix is shipped with  2.12.1 
Flexibility is added to control the icon size with `ShareSheetStyle#setIconSize() `

**Comment by EvangelosG** [comment]
@ming13 Let us know if you have any issues with app icon sizing :) Thank you again for reporting this!
